lib/widgets/profile.dart

```diff
@@ -0,0 +1,261 @@
+    if (choiceItem == null) return null;
+    return _buildTextWidget(choiceItem.text);
+  }
+
+  Widget? _buildExternalAccountField(BuildContext context, String value, CustomProfileField realmField) {
+    // TODO(server): This is undocumented.  See chat thread:
+    //   https://chat.zulip.org/#narrow/stream/378-api-design/topic/external.20account.20custom.20profile.20fields/near/1387213
+    final store = PerAccountStoreWidget.of(context);
+    final CustomProfileFieldExternalAccount externalAccountFieldData;
+    try {
+      final realmFieldData = jsonDecode(realmField.fieldData);
+      externalAccountFieldData = CustomProfileFieldExternalAccount.fromJson(realmFieldData);
+    } on FormatException {
+      return null;
+    } on TypeError {
+      return null;
+    }
+    final urlPattern = externalAccountFieldData.urlPattern ??
+      store.realmDefaultExternalAccounts[externalAccountFieldData.subtype]?.urlPattern;
+    if (urlPattern == null) return null;
+    final url = urlPattern.replaceFirst("%(username)s", value);
+    return _buildLinkWidget(context, url, value);
+  }
+
+  Widget? _buildUserField(BuildContext context, String value, CustomProfileField realmField) {
+    // TODO(server): This is completely undocumented.  The key to
+    //   reverse-engineering it was:
+    //   https://github.com/zulip/zulip/blob/18230fcd9/static/js/settings_account.js#L247
+    final List<int> userIds;
+    try {
+      userIds = (jsonDecode(value) as List<dynamic>).map((e) => e as int).toList();
+    } on FormatException {
+      return null;
+    } on TypeError {
+      return null;
+    }
+    if (userIds.isEmpty) {
+      return null;
+    }
+    return Column(
+      crossAxisAlignment: CrossAxisAlignment.start,
+      children: userIds.map((userId) => _buildUserWidget(context, userId)).toList());
+  }
+
+  Widget _buildTextWidget(String text) {
+    return Padding(
+      padding: const EdgeInsets.symmetric(vertical: 4),
+      child: Text(text));
+  }
+
+  Widget _buildLinkWidget(BuildContext context, String text, String url) {
+    final linkNode = LinkNode(url: url, nodes: [TextNode(text)]);
+    final paragraph = Paragraph(node: ParagraphNode(nodes: [linkNode], links: [linkNode]));
+    return MouseRegion(
+      cursor: SystemMouseCursors.click,
+      child: GestureDetector(
+        onTap: () => launchContentUrl(context, url),
+        child: paragraph));
+  }
+
+  Widget _buildUserWidget(BuildContext context, int userId) {
+    final store = PerAccountStoreWidget.of(context);
+    final User? user = store.users[userId];
+    if (user == null) {
+      return _buildTextWidget("Unknown user");
+    }
+    return InkWell(
+      onTap: () => Navigator.push(context,
+        ProfilePage.buildRoute(context: context,
+          userId: userId)),
+      child: Padding(
+        padding: const EdgeInsets.symmetric(vertical: 4),
+        child: Row(
+          children: [
+            Avatar(userId: userId, size: 32, borderRadius: 32 / 8),
```

Comment:
The InkWell needs some horizontal padding too ­— the highlighting looks odd at the left edge because it's flush with the avatar. (And if the user's name were just long enough to reach the right edge, I expect that'd look odd too.)